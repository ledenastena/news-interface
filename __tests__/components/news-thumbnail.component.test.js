import React from 'react'
import { render } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import NewsThumbnail from '../../src/components/news-thumbnail/news-thumbnail.component'

it('should render NewsThumbnail correctly', () => {
  const newsObj = {
    title: 'article1',
    url: '123',
    urlToImage: '333',
    description: 'Article one descriprion',
  }
  const { container } = render(
    <MemoryRouter>
      <NewsThumbnail newsObj={newsObj} />
    </MemoryRouter>
  )
  expect(container.firstChild).toMatchSnapshot()
})
