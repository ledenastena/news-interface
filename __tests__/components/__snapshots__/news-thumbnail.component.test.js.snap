// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render NewsThumbnail correctly 1`] = `
<div
  class="news-thumbnail-container col-l-4 col-m-6 col-s-6"
  data-testid="news-thumbnail-container"
>
  <div
    class="news-thumbnail-title"
  >
    article1
  </div>
  <div
    class="loading-image-container"
  >
    <img
      alt="article1"
      class="news-thumbnail-image"
      src="333"
    />
    <div
      class="loading-ring"
      data-testid="loading-ring"
    />
  </div>
  <div
    class="news-thumbnail-description"
  >
    Article one descriprion
  </div>
  <a
    class="read-more-link"
    href="/article"
  >
    More &gt;
  </a>
</div>
`;
